File: app/src/main/java/com/homewise/android/HomeWiseApplication.java
package com.homewise.android;

import android.app.Application;
import android.content.Context;

import com.crashlytics.android.Crashlytics;
import com.homewise.android.data.AppDataManager;
import com.homewise.android.injection.component.ApplicationComponent;
import com.homewise.android.injection.component.DaggerApplicationComponent;
import com.homewise.android.injection.module.ApplicationModule;

import javax.inject.Inject;

import io.fabric.sdk.android.Fabric;

/**
 * Created by @raj on 26/12/17.
 */
public class HomeWiseApplication extends Application{

    private ApplicationComponent applicationComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        applicationComponent = DaggerApplicationComponent
                .builder()
                .applicationModule(new ApplicationModule(this))
                .build();
        applicationComponent.inject(this);

    }

    public ApplicationComponent getComponent() {
        return applicationComponent;
    }

    public void setComponent(ApplicationComponent applicationComponent) {
        this.applicationComponent = applicationComponent;
    }

    public static HomeWiseApplication get(Context context) {
        return (HomeWiseApplication) context.getApplicationContext();
    }


}

File: app/src/main/java/com/homewise/android/HomeWiseApplication.kt
package com.homewise.android

import android.app.Application
import android.content.Context
import com.crashlytics.android.Crashlytics
import com.homewise.android.injection.component.ApplicationComponent
import com.homewise.android.injection.component.DaggerApplicationComponent
import com.homewise.android.injection.module.ApplicationModule
import io.fabric.sdk.android.Fabric

/**
 * Created by @raj on 26/12/17.
 */
class HomeWiseApplication : Application() {

    private val mApplicationComponent : ApplicationComponent by lazy {
        DaggerApplicationComponent
                .builder()
                .applicationModule(ApplicationModule(this))
                .build()
    }

    override fun onCreate() {
        super.onCreate()
        if (!BuildConfig.DEBUG) {
            Fabric.with(this, Crashlytics())
        }

        mApplicationComponent.inject(this)
    }

    fun getApplictionComponent(): ApplicationComponent {
        return mApplicationComponent
    }

    companion object {
        fun get(context: Context) : HomeWiseApplication{
            return context.applicationContext as HomeWiseApplication
        }
    }
}